### ISSUE TITLE
qase-cucumber integration issue.

### ISSUE THREAD

**Issue by saiyasaswi** [qase-tms/qase-java#27]
as per GitHub documentation, I have added the below plugin in my pom.xml file.

<plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-surefire-plugin</artifactId>
                <version>3.0.0-M4</version>
                <configuration>
                    <argLine>
                        -Dcucumber.options="--add-plugin", "io.qase.cucumber4.QaseEventListener"
                    </argLine>
                </configuration>
            </plugin>

 when I do mvn clean install I am getting the error below error. can someone please help me with what am I missing?

Please refer to D:\ym_automation\ym\ym-automation\web\target\surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was cmd.exe /X /C "C:\jdk-15\bin\java -Dcucumber.options=--add-plugin, io.qase.cucumber4.QaseEventListener -jar C:\Users\LENOVA\AppData\Local\Temp\surefire15401725191437723890\surefirebooter14997335093945692659.jar C:\Users\LENOVA\AppData\Local\Temp\surefire15401725191437723890 2021-12-21T18-34-07_070-jvmRun1 surefire12901265925089891562tmp surefire_03737818991806075674tmp"
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 1
[ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was cmd.exe /X /C "C:\jdk-15\bin\java -Dcucumber.options=--add-plugin, io.qase.cucumber4.QaseEventListener -jar C:\Users\LENOVA\AppData\Local\Temp\surefire15401725191437723890\surefirebooter14997335093945692659.jar C:\Users\LENOVA\AppData\Local\Temp\surefire15401725191437723890 2021-12-21T18-34-07_070-jvmRun1 surefire12901265925089891562tmp surefire_03737818991806075674tmp"
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 1

**Comment by savkk** [comment]
Could you write which JDK build you are using?
You can try to use maven-surefire-plugin 2.22.2 or 3.0.0-M5

**Comment by saiyasaswi** [comment]
i am using JDK build C:\jdk-15

in the above error message also pls observe its showing C:\jdk-15 version

I tried using maven-surefire-plugin 2.22.2 or 3.0.0-M5. getting same error.

**Comment by savkk** [comment]
I can't reproduce this error at AdoptOpenJDK (build 15.0.2+7), and I want to know which JDK build are you using? 

Type `java -version` in the console

**Comment by saiyasaswi** [comment]
my java version is:-

openjdk version "15" 2020-09-15
OpenJDK Runtime Environment (build 15+36-1562)
OpenJDK 64-Bit Server VM (build 15+36-1562, mixed mode, sharing)

**Comment by saiyasaswi** [comment]
In my pom.xml i have two configurations having both argLine and suiteXmlFiles configured like below and this is causing the error

any suggestion? 

<plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-surefire-plugin</artifactId>
                <version>3.0.0-M4</version>
                <configuration>
                    <argLine>
                        -Dcucumber.options="--add-plugin", "io.qase.cucumber4.QaseEventListener"
                    </argLine>
                    <suiteXmlFiles>
                        <suiteXmlFile>src/test/resources/suite/testng.xml</suiteXmlFile>
                    </suiteXmlFiles>
                </configuration>
            </plugin>

**Comment by savkk** [comment]
Could you try to use qase-cucumber4-jvm 2.0.1? It can help you

**Comment by saiyasaswi** [comment]
below is the configuration, still the same issue occurring. if i comment out either argLine or suiteXmlFiles then . mvn clean install is running successfully. 

I need suiteXmlFiles tag also, to hit my tests suite. 

pls suggest.

         <dependency>
            <groupId>io.qase</groupId>
            <artifactId>qase-cucumber4-jvm</artifactId>
            <version>2.0.1</version>
        </dependency>

<plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-surefire-plugin</artifactId>
                <version>3.0.0-M4</version>
                <configuration>
                    <argLine>
                        -Dcucumber.options="--add-plugin", "io.qase.cucumber4.QaseEventListener"
                    </argLine>
                    <suiteXmlFiles>
                        <suiteXmlFile>src/test/resources/suite/testng.xml</suiteXmlFile>
                    </suiteXmlFiles>
                </configuration>
            </plugin>

**Comment by savkk** [comment]
could you try to replace argLine to `-Dcucumber.options="--add-plugin io.qase.cucumber4.QaseEventListener"`?

**Comment by saiyasaswi** [comment]
above command worked. I don't see any error when I do mvn clean install
thanks for the command
but in git hub actions I see below error

![image](https://user-images.githubusercontent.com/74891825/147780263-f56593dd-a073-4ab9-b649-b9a36e77b464.png)

**Comment by saiyasaswi** [comment]
for us, qase migration is stopped because of this issue

pls suggest any solution for the above issue?

**Comment by savkk** [comment]
try to manually add the okhttp3 dependency

```
        <dependency>
            <groupId>com.squareup.okhttp3</groupId>
            <artifactId>okhttp</artifactId>
            <version>4.9.2</version>
        </dependency>
```

**Comment by saiyasaswi** [comment]
added above mentioned dependency in the project pom. getting same error
pls suggest
![image](https://user-images.githubusercontent.com/74891825/148778808-14ecb9c0-38c5-4407-8033-129b028c6e50.png)

**Comment by savkk** [comment]
I think you have a conflict with dependencies. 
I have created a simple project with Cucumber4 and Qase plugin -https://github.com/savkk/qase_cucumber4.
Test run in GitActions finished successes. Results appeared in Qase.

Could you check OkHttp in the transitive dependencies of the project?

**Comment by saiyasaswi** [comment]
@savkk you are right. the issue is related to transitive dependencies of the project because of my project's spring boot dependencies. that's where the conflict came in.
after a bunch of upgrades and some dependencies exclusions, I got web results posted back to QASE. 
thanks so much for the continuous support. Now Integration in the perspective of results is DONE for both TestNG and Cucumber.


I have a question related to test case RUN. let's say I have 10 test cases in QASE, I have triggered a RUN selecting only 3 test cases. how do I make sure only those 3 test cases run in GIT Hub actions.
How do I make sure I run only failed test cases re-run based on results from QASE.
do we have any documentation related to this type of configuration?
